### PR TITLE
Use musl and libc-test submodules

### PR DESCRIPTION
**Pull request by mingweishih**
This PR makes the musl libc and libc-test as submodules instead of caching them as part of OE. The version of both submodules remain the same as the previously cached ones.  The future update to the musl can be directly on the submodules.

Signed-off-by: Ming-Wei Shih <mishih@microsoft.com>

### CODE REVIEW

**Comment by mingweishih** [comment]
bors try

File: 3rdparty/musl/README.md [hunk 1]
```diff
@@ -1,21 +1,42 @@
...
+
+- musl/
+  The clone of musl libc that is included as a git submodule, which points to
+  a recent release tag. To update the submodule, we use the following procedure:
+  - Checkout the tag that we want to update to.
+    ```
+    cd musl
+    git checkout <new tag>
+    ```
+  - Check the diff between the tag and the previous tag and update the <openenclave-root>/libc/ correspondingly.
```
Comment (CodeMonkeyLeet) [inline 3rdparty/musl/README.md:15]: Can you also document the current tag versions that the musl and libc-test projects are currently synced to?

**Review by CodeMonkeyLeet — APPROVED** [review]
LGTM

File: 3rdparty/musl/README.md (hunk 1 above)
Comment (mingweishih) [reply 3rdparty/musl/README.md:16]: Updated. We need to make sure updating the docs when moving forward 😄

**Comment by mingweishih** [comment]
bors try

**Review by anakrish — APPROVED** [review]
LGTM.

I so wish.gitmodules had some way of indicating the tag or commit hash or a comment :(

**Comment by mingweishih** [comment]
> LGTM.
> 
> I so wish.gitmodules had some way of indicating the tag or commit hash or a comment :(

I hope so too. It would be really helpful.

**Comment by radhikaj** [comment]
Awesome improvement! Thanks Ming-Wei!

**Pull request merged**